opengrok-tools/src/main/python/opengrok_tools/scm/git.py

```diff
@@ -39,6 +39,15 @@ def __init__(self, logger, path, project, command, env, hooks, timeout):
         if not self.command:
             raise RepositoryException("Cannot get git command")
 
+        # The incoming() check relies on empty output so configure
+        # the repository first to avoid getting extra output.
+        git_command = [self.command, "config", "--local", "pull.ff", "only"]
+        cmd = self.getCommand(git_command, work_dir=self.path,
+                              env_vars=self.env, logger=self.logger)
+        cmd.execute()
+        if cmd.getretcode() != 0 or cmd.getstate() != Command.FINISHED:
+            cmd.log_error("failed to perform pull")
```

Comment:
How about `"failed to configure git pull.ff"`?

Reply:
fixed